common/src/main/java/org/apache/sedona/common/raster/PixelFunctions.java

```diff
@@ -77,12 +85,42 @@ public static Geometry getPixelAsPoint(GridCoverage2D raster, int colX, int rowY
         }
         return GEOMETRY_FACTORY.createPoint(pointCoord);
     }
-    public static List<Double> values(GridCoverage2D rasterGeom, List<Geometry> geometries, int band) throws TransformException {
+
+    public static List<Double> values(GridCoverage2D rasterGeom, int[] xCoordinates, int[] yCoordinates, int band) throws TransformException {
+        RasterUtils.ensureBand(rasterGeom, band); // Check for invalid band index
         int numBands = rasterGeom.getNumSampleDimensions();
-        if (band < 1 || band > numBands) {
-            // Invalid band index. Return nulls.
-            return geometries.stream().map(geom -> (Double) null).collect(Collectors.toList());
+
+        double noDataValue = RasterUtils.getNoDataValue(rasterGeom.getSampleDimension(band - 1));
+        List<Double> result = new ArrayList<>(xCoordinates.length);
+        double[] pixelBuffer = new double[numBands];
+
+        for (int i = 0; i < xCoordinates.length; i++) {
+            double x = xCoordinates[i];
+            double y = yCoordinates[i];
+
+            GridCoordinates2D gridCoord = new GridCoordinates2D((int) x, (int) y);
```

Comment:
Is there a reason to convert the int coordinate to double and then back to int?

Reply:
Fixed. That was a remanent of my previous approach. Thanks.